libraries/rush-lib/src/cli/RushXCommandLine.ts

```diff
@@ -44,138 +54,170 @@ interface IRushXCommandLineArguments {
-        process.cwd()
-      );
-      if (!packageJsonFilePath) {
-        // eslint-disable-next-line no-console
-        console.log(colors.red('This command should be used inside a project folder.'));
-        // eslint-disable-next-line no-console
-        console.log(
-          `Unable to find a package.json file in the current working directory or any of its parents.`
-        );
-        return;
+      const eventHooksManager: EventHooksManager | undefined = rushConfiguration
+        ? new EventHooksManager(rushConfiguration)
+        : undefined;
+
+      const suppressHooks: boolean = process.env[EnvironmentVariableNames._RUSH_RECURSIVE_RUSHX_CALL] === '1';
+      const attemptHooks: boolean = !suppressHooks && !args.help;
+      if (attemptHooks) {
+        try {
+          eventHooksManager?.handle(Event.preRushx, args.isDebug, args.ignoreHooks);
+        } catch (error) {
+          // eslint-disable-next-line no-console
+          console.error(colors.red('PreRushx hook error: ' + (error as Error).message));
+        }
+      }
+      // Node.js can sometimes accidentally terminate with a zero exit code  (e.g. for an uncaught
+      // promise exception), so we start with the assumption that the exit code is 1
+      // and set it to 0 only on success.
+      process.exitCode = 1;
+      const options: ILaunchRushXInternalOptions = {
+        isManaged,
+        rushConfiguration,
+        args
+      };
+      RushXCommandLine._launchRushXInternal(options);
+      if (attemptHooks) {
+        try {
+          eventHooksManager?.handle(Event.postRushx, args.isDebug, args.ignoreHooks);
+        } catch (error) {
+          // eslint-disable-next-line no-console
+          console.error(colors.red('PostRushx hook error: ' + (error as Error).message));
+        }
       }
 
-      if (rushConfiguration && !rushConfiguration.tryGetProjectForPath(process.cwd())) {
-        // GitHub #2713: Users reported confusion resulting from a situation where "rush install"
-        // did not install the project's dependencies, because the project was not registered.
-        // eslint-disable-next-line no-console
-        console.log(
-          colors.yellow(
-            'Warning: You are invoking "rushx" inside a Rush repository, but this project is not registered in rush.json.'
-          )
-        );
+      // Getting here means that we are all done with no major errors
+      process.exitCode = 0;
+    } catch (error) {
+      if (error instanceof ProcessError) {
+        process.exitCode = error.exitCode;
+      } else {
+        process.exitCode = 1;
       }
+      // eslint-disable-next-line no-console
+      console.error(colors.red('Error: ' + (error as Error).message));
+    }
+  }
 
-      const packageJson: IPackageJson = packageJsonLookup.loadPackageJson(packageJsonFilePath);
+  public static _launchRushXInternal(options: ILaunchRushXInternalOptions): void {
```

Comment:
With your current refactoring,  `_launchRushXInternal()` is now entirely private to `RushXCommandLine`, so we should make it `private` (or merge it with  `launchRushX()`).

Reply:
fixed